### PR TITLE
Fix oppushdata overflow bug

### PR DESCRIPTION
**Pull request by caevv**
fix panic:
```panic: runtime error: slice bounds out of range [3:2]```

### CODE REVIEW

**Comment by jadwahab** [comment]
Thanks for raising this @caevv. From an initial glance, I don't see how your PR changes fix the issue raised. Can you give us some more information regarding the panic situation? Input values so that we can reproduce the panic error you mention above would be great.

**Comment by caevv** [comment]
> Thanks for raising this @caevv. From an initial glance, I don't see how your PR changes fix the issue raised. Can you give us some more information regarding the panic situation? Input values so that we can reproduce the panic error you mention above would be great.

You are right, there is more to it than that. The size of that `b` seems to be bigger than what uint16 can hold.

I'm using this with tx id: `cba1d7e24dd31cb9cb5d029b825b081e64c65d631ee8a71653a01210da7e1034`
```
	for _, out := range tx.Outputs {
		_, err := out.LockingScript.ToASM()
		if err != nil {
			fmt.Println(err)
		}
	}
```

panics on `LockingScript.ToASM`:
```
panic: runtime error: slice bounds out of range [3:2]

goroutine 1 [running]:
github.com/libsv/go-bt/bscript.DecodeParts(0xc00039214e, 0x1001c, 0x11eb2, 0x0, 0x0, 0x0, 0x0, 0x0)
        .../github.com/libsv/go-bt/bscript/oppushdata.go:109 +0x1054
github.com/libsv/go-bt/bscript.(*Script).ToASM(0xc00000e240, 0x0, 0x0, 0x0, 0x0)
        .../github.com/libsv/go-bt/bscript/script.go:200 +0xaa
```

File: bscript/oppushdata.go [hunk 1]
```diff
@@ -90,37 +90,51 @@ func DecodeParts(b []byte) ([][]byte, error) {
...
+			b = b[l:]
 
 		case OpPUSHDATA2:
 			if len(b) < 3 {
 				return r, errors.New("not enough data")
 			}
-			l := binary.LittleEndian.Uint16(b[1:])
-			if len(b) < int(3+l) {
+
+			l := int(binary.LittleEndian.Uint16(b[1:]))
```
Comment (theflyingcodr) [inline bscript/oppushdata.go:110]: Would probably be better to use  binary.LittleEndian.Uint64(b[1:])  here and it should resolve this issue without needing to convert to int etc

**Comment by ordishs** [comment]
Regarding this last comment about using binary.LittleEndian.Uint64(), this is incorrect.  If you did make this change, we would read 8 bytes from the script rather than the 2 that we need for a uint16.

Happy to explain further, if required.

Regards

Simon

**Comment by theflyingcodr** [comment]
Good point @ordishs; I think the other approach is probably best then.

Would you be able to add a unit test in with the failing script to show that the fix works and to ensure if this code is changed in future that the test catches it.

**Pull request merged**